expression/column.go

```diff
@@ -55,15 +55,15 @@ func (col *CorrelatedColumn) IsCorrelated() bool {
 }
 
 // Decorrelate implements Expression interface.
-func (col *CorrelatedColumn) Decorrelate(schema Schema) Expression {
-	if schema.GetColumnIndex(&col.Column) == -1 {
+func (col *CorrelatedColumn) Decorrelate(cols []*Column) Expression {
```

Comment:
I think it's ok for these function to receive Schema directly. Only GetColumnIndex is needed.